### DISCUSSION TITLE
How can I detect if a variable is exist in a silo file

### DISCUSSION THREAD

**Discussion by lwJi** [LLNL/Silo#429] [Q&A] [answered]
How can I detect if a variable is exist in a silo file, before I call `DBGetQuadvar`?

**Comment by markcmiller86** [comment, marked as answer]
[DBInqVarExists()](https://silo.readthedocs.io/generic.html#dbinqvarexists)

**Reply by markcmiller86** [reply]
If you just want to see everything that is available, use [`DBGetToc()`](https://silo.readthedocs.io/files.html#dbgettoc)

**Reply by lwJi** [reply]
Thanks Mark. I appreciate it.

**Reply by lwJi** [reply]
By the way, is there an easy way to get the number of .silo files inside a .silo.dir? Thanks

**Reply by markcmiller86** [reply]
I am not entirely sure what you are asking here. But, if you are talking multi-block situations where you have maybe a *root* file like `foo_000000_root.silo` and that file contains multi-block objects that then point at blocks in files in a *companion* directory, `foo_000000.dir` and you would like to know how many files are in that directory given a multi-block object read from `foo_000000_root.silo`, then NO, there isn't anything in Silo library that would help with that. But its a relatively simple operation to compute in C++ or Python. In C++. I would just iterate over all the (string) members of a multi-block object, taking everything before the colon character and add them to a *set* object. When done, the size of the set object is the number of unique files involved.

**Reply by lwJi** [reply]
Thanks Mark, I know want to do now ;) Thanks for your help.
